src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -185,9 +185,15 @@ class OrderedRVD(
     if (!shuffle && maxPartitions >= n)
       return this
     if (shuffle) {
-      val shuffled = rdd.coalesce(maxPartitions, shuffle = true)
-      val ranges = OrderedRVD.calculateKeyRanges(typ, OrderedRVD.getPartitionKeyInfo(typ, OrderedRVD.getKeys(typ, shuffled)), shuffled.getNumPartitions)
-      OrderedRVD.shuffle(typ, new OrderedRVDPartitioner(typ.partitionKey, typ.kType, ranges), shuffled)
+      val shuffled = crdd.coalesce(maxPartitions, shuffle = true)
```

Comment:
I know it was this way before too, but I'm not seeing the reason for the `coalesce`. Why not compute the new ranges directly from the original OrderedRVD, and only do the one shuffle instead of two? (If you think this is too far out of the scope of the PR, that's fine.)

Reply:
cc: @cseed @tpoterba

I'm not sure why the coalesce is necessary before calling `OrderedRVD.getKeys`.

Reply:
This is a relic from optimizing the common use case -- coalesce to a small number of huge partitions. In this case, it's far cheap to shuffle the small filtered data than to recompute the filter twice. This doesn't matter much though right now.

Reply:
I don't think I understand, but I'm happy to remove `shuffled` and replace its uses with `crdd` if that sounds right to you.

Reply:
@tpoterba ping

Reply:
I don't particularly care, either one is probably ok